engines/director/rte.cpp

```diff
@@ -51,35 +64,57 @@ RTE2::RTE2(Cast *cast, Common::SeekableReadStreamEndian &stream) : _cast(cast) {
 	bpp = stream.readUint16BE();
 	int checkMax = (1 << bpp) - 1;
 	debugC(5, kDebugLoading, "RTE2: width: %d, height: %d, bpp: %d", width, height, bpp);
-	alphaMap.resize(width*height, 0);
+	// Create a 32-bit alpha surface for the decoded image
+	_surface = new Graphics::Surface();
+	_surface->create((int16)width, (int16)height, Graphics::PixelFormat(4, 8, 8, 8, 8, 24, 16, 8, 0));
+	uint8 r = 0;
+	uint8 g = 0;
+	uint8 b = 0;
+	uint8 a = 0;
 	for (int y = 0; y < height; y++) {
 		for (int x = 0; x < width;) {
 			uint32 pos = stream.pos();
 			byte check = stream.readByte();
 			if (check == 0x1f) {
-				debugC(9, kDebugLoading, "(%d, %d): %x -> color %d %d %d", x, y, pos, stream.readByte(), stream.readByte(), stream.readByte());
+				r = stream.readByte();
+				g = stream.readByte();
+				b = stream.readByte();
+				debugC(9, kDebugLoading, "[%04x] (%d, %d): color %d %d %d", pos, x, y, r, g, b);
 				continue;
 			}
+			a = ((uint32)check*0xff/((1 << bpp) - 1));
 			if (check == 0 || check == checkMax) {
 				byte count = stream.readByte();
-				debugC(9, kDebugLoading, "(%d, %d): %x -> %02x %02x", x, y, pos, check, count);
+				debugC(9, kDebugLoading, "[%04x] (%d, %d): %02x, count %d", pos, x, y, check, count);
 				if (count == 0x00 && check == 0x00) {
-					// end of line
+					// end of line, fill the remaining colour
+					a = 0;
```

Comment:
Would not that make alpha 0, e.g. full transparency? Shouldn't it be `0xff`?

Reply:
Full transparency is correct. The end result is an RGBA surface containing the letters on a transparent background, and if the RLE skips to the end of a horizontal line there's no more letter data for that line.